jsapp/js/attachments/AttachmentActionsDropdown.tsx

```diff
@@ -0,0 +1,143 @@
+import { FocusTrap, Group, Menu, Modal, Stack } from '@mantine/core'
+import { useState } from 'react'
+import ActionIcon from '#/components/common/ActionIcon'
+import Button from '#/components/common/ButtonNew'
+import Icon from '#/components/common/icon'
+import { userHasPermForSubmission } from '#/components/permissions/utils'
+import { QuestionTypeName } from '#/constants'
+import type { AssetResponse, SubmissionResponse } from '#/dataInterface'
+import { FeatureFlag, useFeatureFlag } from '#/featureFlags'
+import { downloadUrl, notify } from '#/utils'
+import { useRemoveAttachment } from './attachmentsQuery'
+
+interface AttachmentActionsDropdownProps {
+  asset: AssetResponse
+  submissionData: SubmissionResponse
+  attachmentId: number
+  /**
+   * Being called after attachment was deleted succesfully. Is meant to be used
+   * by parent component to reflect this change in the data it holds, and
+   * possibly in other places in UI.
+   */
+  onDeleted: () => void
+}
+
+/**
+ * Displays a "…" button that opens a dropdown with some actions available for
+ * provided attachment. Delete option would display a safety check modal.
+ */
+export default function AttachmentActionsDropdown(props: AttachmentActionsDropdownProps) {
+  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState<boolean>(false)
+  const [isDeletePending, setIsDeletePending] = useState<boolean>(false)
+  const removeAttachmentMutation = useRemoveAttachment(props.asset.uid, props.submissionData['meta/rootUuid'])
+  const isFeatureEnabled = useFeatureFlag(FeatureFlag.removingAttachmentsEnabled)
+
+  // TODO: remove this when feature is ready. For now we hide the whole thing by not rendering anything.
+  if (!isFeatureEnabled) {
```

Comment:
A bit of confusion, in the [spec](https://www.notion.so/kobotoolbox/Removing-attachments-8e2eb538680e427fa14173ec523c58e2?p=cb1e69c1f6ec4dd7b4586c7a1c45b5e7&pm=s) it's written that
> if feature flag is not enabled, this dropdown will only have “Download” option

but here we don't render the entire thing. It does say that for if the attachment is deleted then this should be the case.

Reply:
Right! Fixed this. Now only "Delete" button is hidden behind feature flag

Reply:
See https://github.com/kobotoolbox/kpi/pull/5597